src/components/views/dialogs/TermsDialog.js

```diff
@@ -0,0 +1,209 @@
+            checked={this.props.checked}
+        />;
+    }
+}
+
+export default class TermsDialog extends React.PureComponent {
+    static propTypes = {
+        /**
+         * Array of [Service, policies] pairs, where policies is the response from the
+         * /terms endpoint for that service
+         */
+        policiesAndServicePairs: PropTypes.array.isRequired,
+
+        /**
+         * urls that the user has already agreed to
+         */
+        agreedUrls: PropTypes.arrayOf(PropTypes.string),
+
+        /**
+         * Called with:
+         *     * success {bool} True if the user accepted any douments, false if cancelled
+         *     * agreedUrls {string[]} List of agreed URLs
+         */
+        onFinished: PropTypes.func.isRequired,
+    }
+
+    constructor(props) {
+        super();
+        this.state = {
+            // url -> boolean
+            agreedUrls: {},
+        };
+        for (const url of props.agreedUrls) {
+            this.state.agreedUrls[url] = true;
+        }
+    }
+
+    _onCancelClick = () => {
+        this.props.onFinished(false);
+    }
+
+    _onNextClick = () => {
+        this.props.onFinished(true, Object.keys(this.state.agreedUrls).filter((url) => this.state.agreedUrls[url]));
+    }
+
+    _nameForServiceType(serviceType, host) {
+        switch (serviceType) {
+            case Matrix.SERVICE_TYPES.IS:
+                return <div>{_t("Identity Server")}<br />({host})</div>;
+            case Matrix.SERVICE_TYPES.IM:
+                return <div>{_t("Integrations Manager")}<br />({host})</div>;
+        }
+    }
+
+    _summaryForServiceType(serviceType, docName) {
+        switch (serviceType) {
+            case Matrix.SERVICE_TYPES.IS:
+                return <div>
+                    {_t("Find others by phone or email")}
+                    <br />
+                    {_t("Be found by phone or email")}
+                    {docName !== null ? <br /> : ''}
+                    {docName !== null ? '('+docName+')' : ''}
+                </div>;
+            case Matrix.SERVICE_TYPES.IM:
+                return <div>
+                    {_t("Use bots, bridges, widgets and sticker packs")}
+                    {docName !== null ? <br /> : ''}
+                    {docName !== null ? '('+docName+')' : ''}
+                </div>;
+        }
+    }
+
+    _onTermsCheckboxChange = (url, checked) => {
+        this.setState({
+            agreedUrls: Object.assign({}, this.state.agreedUrls, { [url]: checked }),
+        });
+    }
+
+    render() {
+        const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
+        const DialogButtons = sdk.getComponent('views.elements.DialogButtons');
+
+        const rows = [];
+        for (const policiesAndService of this.props.policiesAndServicePairs) {
+            const parsedBaseUrl = url.parse(policiesAndService.service.baseUrl);
+
+            const PolicyValues = Object.values(policiesAndService.policies);
```

Comment:
Why does this start with a capital letter? I would expect `policyValues` I think...

Reply:
random. fixed!